glnx-macros.h

```diff
@@ -111,5 +111,29 @@ G_BEGIN_DECLS
 
 #endif /* ifndef G_IN_SET */
 
+/* Cleaner one-liner method to iterate over a GHashTable without polluting your
+ * scope (or having to introduce a new one). I.e. rather than
+ *
+ *   gpointer k, v;
+ *   GHashTableIter it;
+ *   g_hash_table_iter_init (&it, ht);
+ *   while (g_hash_table_iter_next (&it, &k, &v)) {
+ *      ...
+ *   }
+ *
+ * you can simply do
+ *
+ *   GLNX_HASH_TABLE_FOREACH(my_hash_table, it, k, v) {
+ *     // use it, k, and v as usual
+ *   }
+ *
+ * `it`, `k`, and `v` are scoped to within the for loop. All variables are
+ * allocated on the stack, as usual. Though note that `it` is already a pointer
+ * to a GHashTableIter.
+ */
+#define GLNX_HASH_TABLE_FOREACH(ht, it, k, v)                                \
+    for (gpointer k, v, it = ({ gpointer p = alloca(sizeof(GHashTableIter)); \
```

Comment:
Hmm, yeah agreed re. `alloca()`, so maybe `GLNX_HASH_TABLE_ALLOCA_FOREACH` ?

> Actually wait I'm confused...why are we even doing that, can't we just call `g_hash_table_iter_init()` before the `for` loop?

I want everything to be contained within the scope of the loop. The only way to do that is to declare it in the loop declaration itself. Unless we recommend as usage something like

```c
{ GLNX_HASH_TABLE_FOREACH(ht, it, k, v)
  {
    ...
  }
}
```

which is not the end of the world, though clearly a more leaky abstraction. :(

I think it's we're not able to seal it up tight to make it feel like a regular `for` loop, then it's probably not worth it.

Or we could have the `ALLOCA` variant which allocates its own iterator, while the regular one expects the iterator to already be allocated, like systemd?